qiskit/transpiler/synthesis/aqc/__init__.py

```diff
@@ -10,169 +10,54 @@
-        num_qubits=num_qubits,
-        network_layout=layout,
-        connectivity_type=connectivity,
-        depth=depth
-    )
-
-    # Create an optimizer to be used by AQC
-    optimizer = L_BFGS_B()
-
-    # Create an instance
-    aqc = AQC(optimizer)
-
-    # Create a template circuit that will approximate our target circuit
-    approximate_circuit = CNOTUnitCircuit(num_qubits=num_qubits, cnots=cnots)
-
-    # Create an objective that defines our optimization problem
-    approximating_objective = DefaultCNOTUnitObjective(num_qubits=num_qubits, cnots=cnots)
-
-    # Run optimization process to compile the unitary
-    aqc.compile_unitary(
-        target_matrix=unitary,
-        approximate_circuit=approximate_circuit,
-        approximating_objective=approximating_objective
-    )
-
-Now ``approximate_circuit`` is a circuit that approximates the target unitary to a certain
-degree and can be used instead of the original matrix.
-
-This uses a helper function, :obj:`make_cnot_network`.
-
-.. autofunction:: make_cnot_network
-
-One can take advantage of accelerated version of objective function. It implements the same
-mathematical algorithm as the default one ``DefaultCNOTUnitObjective`` but runs several times
-faster. Instantiation of accelerated objective function class is similar to the default case:
-
-    # Create an objective that defines our optimization problem
-    approximating_objective = FastCNOTUnitObjective(num_qubits=num_qubits, cnots=cnots)
-
-The rest of the code in the above example does not change.
-
-References:
-
-    [1]: Liam Madden, Andrea Simonetto, Best Approximate Quantum Compiling Problems.
-        `arXiv:2106.05649 <https://arxiv.org/abs/2106.05649>`_
 """
 
-from .approximate import ApproximateCircuit, ApproximatingObjective
-from .aqc import AQC
-from .aqc_plugin import AQCSynthesisPlugin
-from .cnot_structures import make_cnot_network
-from .cnot_unit_circuit import CNOTUnitCircuit
-from .cnot_unit_objective import CNOTUnitObjective, DefaultCNOTUnitObjective
-from .fast_gradient.fast_gradient import FastCNOTUnitObjective
+import warnings
+
+from qiskit.synthesis.unitary.aqc.approximate import ApproximateCircuit, ApproximatingObjective
+from qiskit.synthesis.unitary.aqc import AQC
+from qiskit.synthesis.unitary.aqc.cnot_structures import make_cnot_network
+from qiskit.synthesis.unitary.aqc.cnot_unit_circuit import CNOTUnitCircuit
+from qiskit.synthesis.unitary.aqc.cnot_unit_objective import (
+    CNOTUnitObjective,
+    DefaultCNOTUnitObjective,
+)
+from qiskit.synthesis.unitary.aqc.fast_gradient.fast_gradient import FastCNOTUnitObjective
+from qiskit.transpiler.passes.synthesis.aqc_plugin import AQCSynthesisPlugin
+
+warnings.warn(
+    "The qiskit.transpiler.synthesis.aqc module is pending deprecation since Qiskit 0.46.0. "
+    "It will be deprecated in a following release, no sooner than 3 months after the 0.46.0 release.",
+    stacklevel=2,
+    category=PendingDeprecationWarning,
```

Comment:
```suggestion
    "The qiskit.transpiler.synthesis.aqc module is deprecated since Qiskit 0.46.0 "
    "and will be removed in Qiskit 1.0.",
    stacklevel=2,
    category=DeprecationWarning,
```